test/ConditionalWaterDrop.ts

```diff
@@ -223,22 +215,22 @@ describe("ConditionalWaterDrop", function () {
     increaseTime(3700);
 
     // Try claim
-    await waterDrops.connect(alice).claim();
+    await expect(waterDrops.connect(alice).claim())
+      .to.emit(waterDrops, "Claimed")
+      .withArgs(alice.address, 1000000);
```

Comment:
@hrishibhattu, you only added a test for 1 of the events, you need to be testing ALL the events.

Reply:
I've now added tests for both of the events that are emitted in `ConditionalWaterDrops.sol`.
1. I've removed the `event(FlowRetrieved)` because events cannot be emitted from view functions.
2. I have not added any events on my own. I have added events as per `WaterDrops.sol`.
3. Currently there are two failing tests in `WaterDrops.ts`. I did not introduce these bugs, and don't have the knowledge to solve them.